app/views/layouts/application.haml

```diff
@@ -79,8 +79,8 @@
                 %li
                   = link_to messages_path do
                     Inbox
-                    - if current_user.messages.unread.exists?
-                      %span.badge.badge-primary= current_user.messages.unread.count
+                    - if Message.unread_count_for(current_user) > 0
+                      %span.badge.badge-primary= Message.unread_count_for(current_user)
```

Comment:
Since we hit the cache for this (an external service call), we should probably just do it once:
```suggestion
                    - unread_count = Message.unread_count_for(current_user)
                    - if unread_count > 0
                      %span.badge.badge-primary= unread_count
```